composer.json

```diff
@@ -23,25 +23,26 @@
         }
     },
     "require": {
-        "php": "^5.5|^7.0",
-        "php-http/httplug": "^1.0",
-        "php-http/message-factory": "^1.0.2",
-        "php-http/client-implementation": "^1.0",
-        "php-http/discovery": "^1.0"
+        "php": "^7.1.3",
+        "php-http/httplug": "^2.0",
+        "php-http/discovery": "^1.4",
+        "php-http/message-factory": "^1.0",
+        "php-http/client-implementation": "^1.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "~4.0",
-        "php-http/guzzle6-adapter": "^1.0",
-        "php-http/message": "^1.0",
+        "phpunit/phpunit": "^7.5",
+        "cache/filesystem-adapter": "^1.0",
         "psr/cache": "^1.0",
```

Comment:
In the constructor you have:

`public function __construct(ExchangeRateServiceContract $service, CacheItemPoolInterface $cacheItemPool = null`

If I do not have `psr/cache` am I able to call the constructor?